zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.group;
+
+import io.camunda.zeebe.db.DbValue;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+
+public class PersistedGroup extends UnpackedObject implements DbValue {
+
+  private final LongProperty groupKeyProp = new LongProperty("groupKey", -1L);
+  private final StringProperty nameProp = new StringProperty("name", "");
```

Comment:
💭 I think this 2 properties shouldn't be optional, but we can change them later on 😄

Reply:
That's a good point. I was going through our persisted objects and we are pretty inconsistent in how we set these values, and even if we store the whole `*Record` object directly, or if we only mirror certain properties.

🤔 Do we have any guides on how to determine what approach should be taken?

Reply:
@nicpuppa I can update them now. Should I also update the `PersistedRole? The properties are also optional there.

Reply:
I have a task on my todo list to review all this properties, I think can be done later. But if you are up to it go ahead 😄 Also `PersistedMapping` and `PersistedTenant` should be checked

Reply:
> 🤔 Do we have any guides on how to determine what approach should be taken?

I'm not sure, but we should have in the persisted object only what is needed, sometimes doesn't make sense to have all the property of the record, if the persisted record is just a wrapper of the record, what's the point to have it 😄

Reply:
I updated the `PersistedGroup`, `PersistedRole`, and `PersistedTenant` classes and removed the default values of the requited properties.

Note that for `PersistedTenant` the `name` property isn't required (at least it isn't used as a column key).